distributed/worker.py

```diff
@@ -2253,70 +2241,24 @@ def release_key(self, key, cause=None, reason=None, report=True):
                 pdb.set_trace()
             raise
 
-    def release_dep(self, dep, report=False):
```

Comment:
It looks like we've replaced `release_dep` calls with `release_keys`. What was the reasoning for this? It'll impact worker plugins

Reply:
There's no longer a distinction between a `dep` and a `Task` which is why I removed it.  But if it's going to impact the plugins I can restore it and have it wrap `release_keys` (or we can update the plugins?)

Reply:
> It'll impact worker plugins

Do we know of any worker plugins that trigger on `release_dep`?

Also, If we're jumping from version 2 to 2020 in the next release then we can also break some things.  Breaking `WorkerPlugin.release_dep` sounds ok to me.

Reply:
> Do we know of any worker plugins that trigger on release_dep?

None come to mind, but there may be user-defined plugins that use `release_dep`

> Breaking WorkerPlugin.release_dep sounds ok to me

Yeah, I think we're treating all the HLG changes + the bump to calver as a major version change, so removing `release_dep` should be okay